### PR TITLE
Added enviroment variables and configured vite.config 

### PR DESCRIPTION
**Pull request by WhosDouble**
things done

- created .env.local file
- configured vite.config to read enviroment variables
- removed the vite_path plugin

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,7 +10,7 @@ lerna-debug.log*
 node_modules
 dist
 dist-ssr
-*.local
+#*.local
```
Comment (dennypenta) [inline .gitignore]: I had bet .env*, because we can have many .env files

File: .env.local [hunk 2]
```diff
@@ -0,0 +1,2 @@
+VITE_API_URL=http://localhost:3000/api
```
Comment (dennypenta) [inline .env.local]: why do we need these 2 variables?

File: vite.config.ts [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
-    tsconfigPaths() 
+    react(), 
   ],
   resolve: {
     alias: {
       "@": path.resolve(__dirname, "./src"),
     }
+  },
+  define: {
+    "process.env.VITE_API_URL": JSON.stringify(process.env.VITE_API_URL),
```
Comment (dennypenta) [inline vite.config.ts]: I don't think we need it

**Review by dennypenta — APPROVED** [review]
the goal to define the variables that we use in auth config for zitadel.
for example ZITADEL_PROJECT_ID=1234 and then use that variable in the auth file.
I don't remember the exact naming, but you got the idea

**Review by dennypenta — CHANGES_REQUESTED** [review]
.

File: .env.local (hunk 2 above)
Comment (WhosDouble) [reply .env.local]: the first one is for making calls to the backend with out hardcoding the url and the second defines the mode of the app like development or production for example

File: vite.config.ts (hunk 3 above)
Comment (WhosDouble) [reply vite.config.ts]: alright i can get rid of this then

File: .env.local (hunk 2 above)
Comment (dennypenta) [reply .env.local]: I doubt we need these variables, we even don't use them

File: .env.local [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+VITE_APP_MODE=development
+
+VITE_ZITADEL_PROJECT_ID=
```
Comment (dennypenta) [inline .env.local]: VITE prefix is extra, we should remove it

File: vite.config.ts (hunk 3 above)
Comment (dennypenta) [reply vite.config.ts:15]: yes

File: .env.local (hunk 4 above)
Comment (WhosDouble) [reply .env.local]: alright ill make the changes

File: vite.config.ts [hunk 5]
```diff
@@ -1,17 +1,14 @@
 import { defineConfig } from 'vite'
 import path from 'path'
 import react from '@vitejs/plugin-react'
-import tsconfigPaths from 'vite-tsconfig-paths'
```
Comment (dennypenta) [inline vite.config.ts:4]: it's nice to delete this dependency if we don't use it, but I can do it later

**Review by dennypenta — APPROVED** [review]

**Pull request merged**